### PR TITLE
feat: user's default currency #P23-385

### PR DESCRIPTION
**Pull request by AzulloCyll**
Task: https://tracker.intive.com/jira/browse/PATRO23-385

What was done: 
- added Currency item in Settings menu
- added atom for currency in store
- made page for Currency which have currency selector and button to save changes
- made currency saves in localStorage
- addNewBudget modal is using defalt currency so value of currency is automaticaly selected
- currency array lifted up to lib/currency.ts

### CODE REVIEW

**Review by monika-klokosinska-intive — APPROVED** [review]

File: apps/web/app/(navigation)/(singleCard)/settings/currency/page.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const setValue = useSetAtom(currencyAtom);
+
+  const translation = {
+    label: t(dict.currency.title),
+    button: t(dict.currency.button),
+  };
+
+  const changeCurrency = (currency: string) => {
+    setValue(currency);
+    localStorage.setItem("currency", currency);
```
Comment (bdeluga) [inline apps/web/app/(navigation)/(singleCard)/settings/currency/page.tsx:42]: I think @Alessiass made useLocalStorage hook, for this kinds of actions. ✏️

File: apps/web/app/(navigation)/(singleCard)/settings/layout.tsx
Comment (bdeluga) [inline apps/web/app/(navigation)/(singleCard)/settings/layout.tsx:1]: Do we need layout for this page, it isn't that complex.

File: apps/web/app/layout.tsx [hunk 2]
```diff
@@ -25,10 +27,12 @@ const queryClient = new QueryClient();
 
 export default function RootLayout({ children }: LayoutProps) {
   const setLanguage = useSetAtom(languageAtom);
+  const setCurrency = useSetAtom(currencyAtom);
 
   useEffect(() => {
     setLanguage((localStorage.getItem("lang") as Language) || "en");
-  }, [setLanguage]);
+    setCurrency(localStorage.getItem("currency") || "USD");
+  }, [setLanguage, setCurrency]);
```
Comment (bdeluga) [inline apps/web/app/layout.tsx:35]: Would useLocalStorage work here ?

**Review by bdeluga — APPROVED** [review]
Couple of things to debate on. Besides that approved ! 🚀

**Review by Vectari — APPROVED** [review]

**Review by Noshi96 — APPROVED** [review]
I agree with @bdeluga about the useLocalStorage part

File: apps/web/app/(navigation)/(singleCard)/settings/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+"use client";
+
+import styled from "styled-components";
+import { LayoutProps } from "app/layout";
+
+const DivStyled = styled.div`
```
Comment (aniasgit) [inline apps/web/app/(navigation)/(singleCard)/settings/layout.tsx:6]: I think this container can be directly on the page and we don't need any additional layout for it.

**Review by aniasgit — APPROVED** [review]
I agree with @bdeluga  about useLocalStorageHook and redundant layout, other than that it works fine.

File: apps/web/app/(navigation)/(singleCard)/settings/currency/page.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      <H1Styled>
+        <SettingsTitle />
+      </H1Styled>
+      <Separator />
+      <InputWrapper>
+        <CurrencySelect
+          onValueChange={changeCurrency}
+          value={value}
+          label={translation.label}
+          id="currencSelector"
```
Comment (wyola) [inline apps/web/app/(navigation)/(singleCard)/settings/currency/page.tsx:56]: small typo in id :) and I think that most of the files here use kebab case for id -> `currency-selector`

PS I merged today the atomic select, so I think you could use it here as well!

**Review by wyola — APPROVED** [review]
Works good! I agree with previous comments and just one, tiny, small thing from me :)

File: apps/web/app/(navigation)/(singleCard)/settings/currency/page.tsx (hunk 1 above)
Comment (AzulloCyll) [reply apps/web/app/(navigation)/(singleCard)/settings/currency/page.tsx]: done

File: apps/web/app/(navigation)/(singleCard)/settings/layout.tsx (hunk 3 above)
Comment (AzulloCyll) [reply apps/web/app/(navigation)/(singleCard)/settings/layout.tsx]: removed redundant layout :)

**Pull request merged**